### PR TITLE
introduce subscription status as a metric

### PR DESCRIPTION
**Pull request by k15r**
**Description**

Changes proposed in this pull request:

- expose subscription status as new metric eventing_ec_subscription_status

**Related issue(s)**

### CODE REVIEW

**Review by NHingerl — DISMISSED** [review]

**Review by NHingerl — APPROVED** [review]

**Review by NHingerl — DISMISSED** [review]

File: components/eventing-controller/controllers/subscription/jetstream/reconciler.go [hunk 1]
```diff
@@ -124,6 +133,35 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
...
+					cc.ConsumerName,
+					r.Backend.GetConfig().JSStreamName)
+			}
+		}
+		for _, dc := range desiredSubscription.Status.Backend.Types {
+			r.collector.RecordSubscriptionStatus(desiredSubscription.Status.Ready,
+				desiredSubscription.Name,
+				desiredSubscription.Namespace,
+				backendType,
+				dc.ConsumerName,
```
[the remark this answers is not in the document]
Comment (k15r) [reply components/eventing-controller/controllers/subscription/jetstream/reconciler.go:159]: this is not about the consumer readiness, but the subscription. so if the subscription is not ready. all its consumer/subscription combinations will not be ready

File: components/eventing-controller/controllers/subscription/eventmesh/reconciler.go [hunk 2]
```diff
@@ -121,6 +126,17 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return r.handleDeleteSubscription(ctx, sub, log)
 	}
 
+	defer func() {
```
[the remark this answers is not in the document]
Comment (k15r) [reply components/eventing-controller/controllers/subscription/eventmesh/reconciler.go:129]: sorry, leftover from a refactoring, will defer just the function to update the metrics

File: components/eventing-controller/controllers/subscription/eventmesh/reconciler.go [hunk 3]
```diff
@@ -274,11 +271,11 @@ func (r *Reconciler) handleDeleteSubscription(ctx context.Context, subscription
 	removeFinalizer(subscription)
 
 	// update subscription CR with changes
-	defer r.collector.RemoveSubscriptionStatus(subscription.Name, subscription.Namespace, backendType, "", "")
 	if err := r.updateSubscription(ctx, subscription, logger); err != nil {
 		return ctrl.Result{}, err
 	}
 
+	r.collector.RemoveSubscriptionStatus(subscription.Name, subscription.Namespace, backendType, "", "")
```
Comment (muralov) [inline components/eventing-controller/controllers/subscription/eventmesh/reconciler.go:278]: 👍🏼

**Review by muralov — DISMISSED** [review]

**Review by muralov — APPROVED** [review]

**Review by NHingerl — DISMISSED** [review]

**Review by NHingerl — APPROVED** [review]

**Review by NHingerl — APPROVED** [review]

**Review by mfaizanse — APPROVED** [review]

**Review by NHingerl — APPROVED** [review]

**Review by mfaizanse — APPROVED** [review]

**Pull request merged**